controllers/common/common.go

```diff
@@ -274,13 +338,13 @@ func GetRepPoliciesInPlacementBinding(
 	// Use this for removing duplicated policies
 	rootPolicyRequest := GetPoliciesInPlacementBinding(ctx, c, pb)
 
-	result := make([]reconcile.Request, 0, len(rootPolicyRequest)*len(decisions))
+	result := make([]reconcile.Request, 0, len(rootPolicyRequest)*len(decisions.ClustersToRollout))
 
 	for _, rp := range rootPolicyRequest {
-		for _, pd := range decisions {
+		for clusterName := range decisions.ClustersToRollout {
```

Comment:
Can we change      
   `decisions, err := GetDecisions(ctx, c, pb) ` 
to 
suggestion: something like
`        rollouts, err := GetRollouts(ctx, c, pb)`